Story/OnlineUIComponents/SpectatorHud.cs

```diff
@@ -53,6 +53,7 @@ public override void Update()
                     RainMeadow.Debug("Shutting down spectator overlay due to another process request");
                     spectatorOverlay.ShutDownProcess();
                     spectatorOverlay = null;
+                    OnlineManager.mePlayer.isActuallySpectating = false;
```

Comment:
we don't restore camera when spectator hud closes, currently player has to spectate themself

better to check if ac being spectated is ours